code/modules/mob/living/carbon/human/human_mob.dm

```diff
@@ -2200,6 +2200,22 @@ Eyes need to have significantly high darksight to shine unless the mob has the X
 	held_gun.afterattack(possible_target, src)
 	visible_message("<span class='danger'>[src] fires [held_gun][isturf(possible_target) ? "" : " towards [possible_target]!"]</span>")
 
+/mob/living/carbon/human/proc/dchat_step(dir)
+	if(length(grabbed_by))
+		resist_grab()
+	step(src, dir)
+
+/datum/component/deadchat_control/human/Initialize(_deadchat_mode, _inputs, _input_cooldown, _on_removal)
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	_inputs["up"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), NORTH)
+	_inputs["down"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), SOUTH)
+	_inputs["left"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), WEST)
+	_inputs["right"] = CALLBACK(parent, TYPE_PROC_REF(/mob/living/carbon/human, dchat_step), EAST)
+
+	return ..()
```

Comment:
this should be in `deadchat_control.dm`. adding it to a 2000+ line file is less than ideal.

Reply:
Ended up moving the new stuff out to a new file, and moving this new datum into `deadchat_control.dm`